service/src/main/java/org/broadinstitute/listener/relay/Utils.java

```diff
@@ -54,26 +53,26 @@ public static void writeCORSHeaders(
   }
 
   public static boolean isValidOrigin(String origin, CorsSupportProperties corsSupportProperties) {
+    if (corsSupportProperties.validHosts().contains("*")) {
+      return true;
+    }
+
     String stringToCheck;
     // We want to strip the protocol.
+    URL url;
     try {
-      URL url = new URL(origin);
-      stringToCheck = url.getHost();
+      url = new URL(origin);
```

Comment:
Quick check - if origin is empty, will this trigger the MalformedURLException? We don't want to, in that case, right? Should we add the empty check above, where we check the wildcard?